floater/utils.py

```diff
@@ -269,9 +269,9 @@ def floats_to_netcdf(input_dir, output_fname,
             time = np.array([ref_time+del_time])
         else:
             time = np.array([np.int32(step_num)])
-        npart = dfcs.npart.values
+        npart = np.int32(dfcs.npart.values)
         var_shape = (1, len(npart))
-        data_vars = {var_name: (['time', 'npart'], dfcs[var_name].values.reshape(var_shape)) for var_name in var_names}
+        data_vars = {var_name: (['time', 'npart'], np.float32(dfcs[var_name].values).reshape(var_shape)) for var_name in var_names}
```

Comment:
better to use `.astype(np.float32)`.

But I'm not sure this should be necessary.